OSSMDOC-705: Update content  for Service Mesh Extension API deprecation (#51803)
openshift/openshift-docs: modules/ossm-extensions-overview.adoc

[NOTE]
====

> When creating new WebAssembly extensions, use the WasmPlugin API. The ServiceMeshExtension API is deprecated as of {SMProductName} version 2.2 and was removed as of {SMProductName} version 2.3.

Reviewer: ```suggestion
When creating new WebAssembly extensions, use the `WasmPlugin` API. The `ServiceMeshExtension` API, which was deprecated in {SMProductName} version 2.2, is removed in {SMProductName} version 2.3.
```

Reviewer: Similar for the following instances.
